subprojects/docs/src/docs/userguide/kotlin_dsl.adoc

```diff
@@ -492,11 +492,78 @@ The example above centrally defines plugin versions in the settings script's plu
 Note that alternatively you could also move the version declarations in the root project `plugins {}` block.
 
 
-// [[sec:collections]]
-// == TODO Working with collections
-// overview & api
-// === Via Kotlin delegated properties
-// === Container scope string invoke extension
+[[sec:containers]]
+== Working with containers
+
+The Gradle build model makes heavy use of containers.
+For example, Gradle core contributes `configurations` which is a container of `Configuration` objects and `tasks` which is a container of `Task` objects ; core plugins such as `java` or `distribution` contribute containers such as `sourceSets` or `distributions`.
+Community plugins also contribute containers, such as `android.buildTypes` by the `android` plugin.
+Gradle plugins create containers using `Project.container<Type>()`.
+
+[TIP]
+--
+Note that if you are configuring existing elements on supported containers you can leverage the type-safe accessors as described in the <<sec:configuring_plugins>> section above.
+--
+
+The Gradle Kotlin DSL provides with uniform ways of interacting with containers.
+We already saw some examples configuring source sets or tasks in the <<sec:configuring_plugins>> section above.
+Let's now look in detail at how to interact with containers using the Gradle Kotlin DSL by using the `tasks` container as an example.
+
+=== Using the containers API
+
+All the containers implement `NamedDomainObjectContainer<DomainObjectType>`, some of them can contain objects of different types and implement `PolymorphicDomainObjectContainer<BaseType>`. The simplest way to interact with containers is through their API.
```

Comment:
:o: _The simplest way to interact with containers is through their API._ instead of?

Maybe the different ways should have been listed before this sentence comes up?

Reply:
I rephrased it a bit. We first show the API then the other ways, no need to enumerate first.